test/metabase/api/field_test.clj

```diff
@@ -770,11 +768,12 @@
                 (is (false? (:json_unfolding field))))
               (testing "nested fields are not present since json unfolding is disabled by default"
                 (is (empty? (nested-fields))))
-              (testing "nested fields are removed when json unfolding is disabled"
+              (testing "nested fields aren't added when json unfolding is enabled, because they never existed"
                 (enable-json-unfolding! true)
+                (is (empty? (nested-fields))))
+              (testing "new nested fields are only added when sync is run again"
                 (sync/sync-database! database)
```

Comment:
This sync is the only time where the manual sync is needed - when the nested JSON fields never existed in the first place.